### PR TITLE
feat: Implement WatchTower

### LINKED ISSUE #104: Implement WatchTower
## Description 📹

**Implement WatchTower as described in the repository [api.starknet.quest](https://github.com/lfglabs-dev/api.starknet.quest).**  
This involves setting up WatchTower for monitoring and handling operations as described in the linked repository's implementation.

## Proposed Actions 🛠️

Here’s a checklist of actions to follow for resolving this issue:

1. **Fork and Create Branch**:  
   Fork the repository and create a new branch using the issue number:

   ```bash
   git checkout -b fix-[issue-number]
   ```

2. **Implement Changes**:  
   - [ ] Review the [api.starknet.quest implementation](https://github.com/lfglabs-dev/api.starknet.quest) for the WatchTower setup.  
   - [ ] Implement similar functionality tailored to the current repository's structure and requirements.
   - [ ] Ensure necessary dependencies are added and documented.

3. **Run Tests and Commit Changes**:  
   Make sure your changes don't break existing functionality and commit with a clear message:

   ```bash
   git commit -m "Fix: Implement WatchTower functionality as per api.starknet.quest"
   ```

## Required 📋

To keep our workflow smooth, please make sure you follow these guidelines:

- **Assignment**: Don't create a pull request if you weren’t assigned to this issue.
- **Timeframe**: Complete the task within **3 business days**.
- **Closing the Issue**: In your PR description, close the issue by writing Close #[issue_id].
- **Review Process**:
  - Once you've submitted your PR, change the label to **"ready for review"**.
  - If changes are requested, address them and then update the label back to **"ready for review"** once done.
- **Testing**: Test your PR locally before pushing, and verify that tests and build are working after pushing.

Thank you for your contribution 🙏

⚠️ WARNING: Failure to follow the requirements above may result in being added to the OnlyDust blacklist, affecting your ability to receive future rewards.

### PR DESCRIPTION
**Pull request by TropicalDog17**
Closes #104

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -32,6 +34,22 @@ async fn main() {
...
     let conf = config::load();
 
+    let logger = logger::Logger::new(&conf.watchtower);
+
+    // Testing logger when server started
+    logger.info(format!(
+        "id_server: starting v{}",
+        env!("CARGO_PKG_VERSION")
+    ));
+    logger.warning(format!(
```
Comment (Marchand-Nicolas) [inline src/main.rs]: Please remove the warning and the severe, we just want to keep the info one

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]
Just one little change

File: src/main.rs (hunk 1 above)
Comment (TropicalDog17) [reply src/main.rs]: Should I remove the logger.warning in `main.rs` or get rid of the `Warning` and `Severe` variant? @Marchand-Nicolas?

Comment (Marchand-Nicolas) [reply src/main.rs]: @TropicalDog17 Remove logger.warning & logger.servere in main.rs. But we keep the warning & severe variants

**Review by Marchand-Nicolas — APPROVED** [review]
Lgtm!

**Pull request merged**